graphql_compiler/compiler/emit_sql.py

```diff
@@ -117,6 +117,9 @@ def _find_used_columns(
             # A recurse implies an outgoing foreign key usage
             child_location_info = ir.query_metadata_table.get_location_info(child_location)
             if child_location_info.recursive_scopes_depth > location_info.recursive_scopes_depth:
+                used_columns.setdefault(get_vertex_path(location), set()).add(
+                    edge.to_column
+                )
```

Comment:
The PR is still draft because I want to spend more time thinking if this is sufficient. Maybe I should explicitly add the primary key as a used column?